### PR TITLE
size_from_dim(0) is like numel() but worse. Don't do it.

### PR DESCRIPTION
**Pull request by ezyang**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#12729 size_from_dim(0) is like numel() but worse. Don't do it.**&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D10415069/)

This may have a dependency on D10380678 if size_from_dim(0)
was required because numel() used to return -1 in some cases.
This is no longer true.

Differential Revision: [D10415069](https://our.internmc.facebook.com/intern/diff/D10415069/)

### CODE REVIEW

**Review by li-roy — APPROVED** [review]

**Comment by ezyang** [comment]
It doesn't look like this caused anything to fail, but just to be safe I won't land this until numel != -1 patch lands.

**Pull request closed without merging**
